user/super/com/google/gwt/emul/java/util/stream/DoubleStream.java

```diff
@@ -162,6 +162,30 @@ public boolean tryAdvance(DoubleConsumer action) {
     return StreamSupport.doubleStream(spliterator, false);
   }
 
+  static DoubleStream iterate(double seed, DoublePredicate hasNext, DoubleUnaryOperator f) {
+    Spliterator.OfDouble spliterator =
+        new Spliterators.AbstractDoubleSpliterator(
+            Long.MAX_VALUE, Spliterator.IMMUTABLE | Spliterator.ORDERED) {
+          private double next = seed;
+          private boolean terminated = false;
+
+          @Override
+          public boolean tryAdvance(DoubleConsumer action) {
+            if (terminated) {
+              return false;
+            }
+            if (!hasNext.test(next)) {
+              terminated = true;
+              return false;
+            }
+            action.accept(next);
+            next = f.applyAsDouble(next);
```

Comment:
when used together with `limit()` or `takeWhile()`, this calls `f` one more time than needed

Reply:
Looks like `*Stream.iterate(seed, f)` is also wrong, has been since initial commit. I'll add tests and get this fixed, thanks for calling it out.

Reply:
Thanks again for calling this out - I updated tests of both old and new `iterate()`, and made the old version delegate to the new one (which will reduce the size impact of using both methods).